### ISSUE TITLE
Gorouter does not handle abrupt termination of client for a streaming api.

### ISSUE THREAD

**Issue by sreenani** [cloudfoundry/gorouter#169]
Thanks for submitting an issue to `gorouter`. We are always trying to improve! To help us, please fill out the following template.

## Issue
We have a streaming api deployed in cf, a curl client has initaited the request and abruptly terminated the connection, gorouter is not able or recognize this and timining out only after request_timeout_in_seconds is reached (15 mins in our case).

## Context
Java rest api deployed in the cf.
If the curl is used to directly hit the api in the dea and port, this as expected.

## Expected result
Gorouter should recognize that client has terminated.

## Current result
Gorouter is not able to recognize the client termination, and streaming api is streaming the data until request_timeout_in_seconds is reached.

**Comment by shashwathi** [comment]
@sreenani : 

Thank you for reporting this issue. I am surprised to hear about this because routing team has Golang streaming [API client](https://github.com/cloudfoundry-incubator/routing-api/blob/master/docs/api_docs.md#subscribe-to-events-for-tcp-routes); when the connection is terminated from the client, gorouter closes the connection immediately. 

Can you provide a sample Java REST API app to reproduce this case? and what is the client to initiate the connection? 

Regards
Shash

**Comment by shashwathi** [comment]
@sreenani  : Closing this issue due to inactivity. Feel free to open another issue if you are still having issue.

Regards
Shash
